Neos.Flow/Classes/Mvc/Routing/Router.php

```diff
@@ -81,6 +53,11 @@ class Router implements RouterInterface
      */
     protected $lastResolvedRoute;
 
+    public function __construct(
+        private readonly Routes $routes
```

Comment:
OK, this is a flaw I'm afraid.
The whole point of the configuration manager injection was to _avoid having to parse the routes for every request_.
The router would only build the route objects if it comes across a route it hasn't seen (i.e. cached) before.
By injecting the parsed routes, we defeat that purpose and this will be a _huge_ performance hit.


But I think we can work around this by adding one layer of abstraction, with the mentioned interface that defers loading & parsing of the routes until requested.